client/src/main/scala/cakesolutions/kafka/KafkaProducerRecord.scala

```diff
@@ -59,6 +61,24 @@ object KafkaProducerRecord {
       case None => new ProducerRecord(topic, value)
     }
 
+  /**
+    * Create a producer record with an optional key.
+    *
+    * @param topic the topic where the record will be appended to
+    * @param key optional key that will be included in the record
+    * @param value the value that will be included in the record
+    * @param headers the record headers
+    * @tparam Key type of the key
+    * @tparam Value type of the value
+    * @return producer record
+    */
+    def apply[Key >: Null, Value](topic: String, key: Option[Key], value: Value,
```

Comment:
Instead of having another flavor of apply, could we have:
def apply[Key >: Null, Value](topic: String, key: Option[Key], value: Value, headers: Option[Seq[Header]] = None)

Reply:
Great point and one that I should have mentioned. The reason I didn't do that was specifically to [avoid breaking backwards binary compatibility](https://github.com/jatcwang/binary-compatibility-guide#dont-adding-parameters-with-default-values-to-methods). If it were a new library would have done it that way. Same for below.